samples/drivers/adc/src/main.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/adc.h>
+
+/*
+ * ADC devicetree nodes have different naming conventions depending on the MCU
+ * vendor.
+ */
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(adc), okay)
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(adc))
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(adc0), okay)
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(adc0))
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(adc1), okay)
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(adc1))
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(adc1_2), okay)
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(adc1_2))
+#elif DT_NODE_HAS_STATUS(DT_NODELABEL(afec0), okay)
+#define ADC_DEVICE_NAME		DT_LABEL(DT_NODELABEL(afec0))
+#else
+#error "Unsupported board: ADC devicetree node not defined"
+#endif
+
+/*
+ * Read sequence of two channels for demonstration (if supported by MCU and
+ * driver)
+ */
+#if defined(CONFIG_ADC_STM32) || \
+	defined(CONFIG_ADC_SAM0)
+#define ADC_NUM_CHANNELS	1
+#else
+#define ADC_NUM_CHANNELS	2
+#endif
+
+#define ADC_1ST_CHANNEL_ID	0
+#define ADC_RESOLUTION		12
+#define ADC_GAIN		ADC_GAIN_1
+#define ADC_REFERENCE		ADC_REF_INTERNAL
```

Comment:
If I'm not wrong not all ADC support the same settings (e.g. resolution may vary, reference...)

Reply:
Ok, will double-check.